components/blocks/BlockComponentName.vue

```diff
@@ -0,0 +1,14 @@
+<script setup lang="ts">
+type Block = {
+	id: string;
+};
+const props = defineProps({
+	data: {
+		type: Object as PropType<Block>,
+		required: true,
+	},
+});
```

Comment:
Lets use TS to define props

```suggestion
interface BlockComponentNameProps {
  data: Block;
}

defineProps<BlockComponentNameProps>();
```